crates/holochain/src/core/workflow/sys_validation_workflow.rs

```diff
@@ -1,4 +1,77 @@
+//! - For a [`DhtOp::RegisterAgentActivity`]
+//!    - Check that the [`Action`] is either a [`Action::Dna`] at sequence number 0, or has a previous action with sequence number strictly greater than 0.
+//!    - If the [`Action`] is a [`Action::Dna`] then check the contained DNA hash matches the DNA hash that sys validation is being run for.
+//!    - Run the [store record checks](#store-record-checks).
+//! - For a [`DhtOp::RegisterUpdatedContent`]
+//!    - The [`Update::original_action_address`] reference to the [`Action`] being updated must point to an [`Action`] that can be found locally. Once the [`Action`] address has been resolved, the [`Update::original_entry_address`] is checked against the entry address that the referenced [`Action`] specified.
+//!    - If there is an [`Entry`] then the [store entry checks](#store-entry-checks) are run.
+//! - For a [`DhtOp::RegisterUpdatedRecord`]
+//!    - The [`Update::original_action_address`] reference to the [`Action`] being updated must point to an [`Action`] that can be found locally. Once the [`Action`] address has been resolved, the [`Update::original_entry_address`] is checked against the entry address that the referenced [`Action`] specified.
+//!    - If there is an [`Entry`] then the [store entry checks](#store-entry-checks) are run.
+//! - For a [`DhtOp::RegisterDeletedBy`]
+//!    - The [`Delete::deletes_address`] reference to the [`Action`] being deleted must point to an [`Action`] that can be found locally. The action being deleted must be a [`Action::Create`] or [`Action::Update`].
+//! - For a [`DhtOp::RegisterDeletedEntryAction`]
+//!    - The [`Delete::deletes_address`] reference to the [`Action`] being deleted must point to an [`Action`] that can be found locally. The action being deleted must be a [`Action::Create`] or [`Action::Update`].
+//! - For a [`DhtOp::RegisterAddLink`]
+//!   - The size of the [`CreateLink::tag`] must be less than or equal to the maximum size that is accepted for this link tag. This is specified in the constant [`MAX_TAG_SIZE`].
+//! - For a [`DhtOp::RegisterRemoveLink`]
+//!   - The [`DeleteLink::link_add_address`] reference to the [`Action`] of the link being deleted must point to an [`Action`] that can be found locally.
+//!
+//! ##### Store record checks
+//!
+//! These checks are run when storing a new action for a [`DhtOp`].
+//!
+//! - Check that the [`Action`] is either a [`Action::Dna`] at sequence number 0, or has a previous action with sequence number strictly greater than 0.
+//! - Checks that the author of the current action is the same as the author of the previous action.
+//! - Checks that the timestamp of the current action is greater than the timestamp of the previous action.
+//! - Checks that the sequence number of the current action is exactly 1 more than the sequence number of the previous action.
+//! - Checks that every [`Action::Create`] or [`Action::Update`] was preceeded by a [`Action::AgentValidationPkg`].
```

Comment:
```suggestion
//! - Checks that every [`Action::Create`] or [`Action::Update`] of an `AgentPubKey` is preceded by an [`Action::AgentValidationPkg`].
```

Reply:
Spot on, but that's not exactly what the code is checking currently. I have tweaked the code on this PR